samples/amcache.py

```diff
@@ -213,6 +218,19 @@ def main(argv=None):
         w.writerow(["timestamp", "timestamp_type", "path", "sha1"])
         for e in sorted(entries, key=lambda e: e.timestamp):
             w.writerow([e.timestamp, e.type, e.entry.path, e.entry.sha1])
+
+    elif args.do_json:
+        for e in ee:
+            document = {}
+            for i in FIELDS:
+                document[i.name] = getattr(e, i.name)
+                if document[i.name] is None:
+                    document[i.name] = "-"
+                elif type(document[i.name]) == datetime.datetime:
+                    document[i.name] = str(document[i.name])
```

Comment:
the logic here makes sense, though i'd recommend using a temporary variable rather than indexing into the `document` dict so many times. perhaps something like:

```
                val = getattr(e, i.name)
                if val is None:
                    val = "-"
                elif type(val) == datetime.datetime:
                    val = str(val)
                document[i.name] = val
```